tests/dao/DaoBlockchain.ts

```diff
@@ -85,6 +95,38 @@ export class DaoBlockchain extends Blockchain {
         return account;
     }
 
+    private async createPeriods() {
+
+      if (this.root === undefined) {
+        throw new Error('Root document has to be created before creating periods');
+      }
+
+      let predecesor = this.root.hash;
+
+      let periodDate = new Date(this.setupDate);
+
+      let numPeriods = this.settings.periodCount;
+
+      for (let i = 0; i < numPeriods; ++i) {
+
+        let label = `Period #${i + 1} of ${numPeriods}`;
+        
+        await this.dao.contract.addperiod({
+          predecessor: predecesor,
+          start_time: periodDate.toISOString().replace('Z', ''),
+          label,
+        });
+
+        let periodDoc = last(getDocumentsByType(this.getDaoDocuments(), 'period'));
+        
+        this.periods.push(new Period(new Date(periodDate), label, periodDoc));
+
+        periodDate.setSeconds(periodDate.getSeconds() + this.settings.periodDurationSeconds);
+
+        predecesor = periodDoc.hash;
```

Comment:
I think this could be optimized (in a separate PR) by somehow creating the periods in a batch or writing the documents directly maybe?

Reply:
We might need to use the snapshot feature for that, but I haven't looked at closely